haystack/preview/telemetry/telemetry.py

```diff
@@ -171,7 +173,12 @@ def send_pipeline_run_event(pipeline: "Pipeline"):
                 for component_name, component in pipeline_description["components"].items():
                     if not component["type"] in components:
                         components[component["type"]] = []
-                    components[component["type"]].append({"name": component_name})
+
+                    telemetry_data = getattr(pipeline.get_component(component_name), "_telemetry_data", {})
+                    if not isinstance(telemetry_data, dict):
+                        telemetry_data = {"error": "_telemetry_data is present but does not contain a dict"}
```

Comment:
This check might be expensive, I would just ignore the payload if it's malformed

Reply:
I still need a try-catch, I'm not sure it's any cheaper